sdk/keyvault/TEST_RESOURCES_README.md

```diff
@@ -0,0 +1,130 @@
+# KeyVault Test Resources
+
+The KeyVault packages provide a common [Azure Resource Manager template][KVARM] to help developers set up the resources needed
+to run all of the integration tests in each of the KeyVault clients. This ARM template is used automatically by the CI pipelines
+that run during pull requests, just as much as it is used to help developers deploy their own resources on their personal
+Azure accounts for manual testing purposes.
+
+If you wish to use this ARM template to deploy resources on your own Azure personal or business account, you can click
+the following deploy button:
+
+[![](http://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-sdk-for-js%2Fmaster%2Fsdk%2Fkeyvault%2Ftest-resources.json)
+
+Keep in mind that deploying these resources constitutes a purchase of Azure services that will be billed to your active account.
+
+## What resources will be created
+
+The [ARM template][KVARM] used by the Deploy Button will create:
+
+- A single standard KeyVault, on the provided location, under the provided tenant, with a name based on the provided "Base Name".
+
+You can read more about KeyVault pricing details [here](https://azure.microsoft.com/en-us/pricing/details/key-vault/).
+
+## The form to deploy the KeyVault resources
+
+Once you click the "Deploy to Azure" button, you will be prompted to fill a form.
+In this guide we will go through the necessary steps to fill this form for the KeyVault clients for JavaScript.
+
+### Logging in
+
+Clicking the button will make Azure attempt to authenticate your Azure account. Make sure to log in using the Azure account that you want to be billed through.
+If you want to **create a new account**, you may follow this guide: [Create an Azure account][CreateAzureAccount].
+
+### The "Subscription" field
+
+The first field you will encounter is the "Subscription" field.
+It will have a drop down menu where you will be able to search for the specific Azure subscription
+you may want to use as the host of the resources you will create.
+
+**This field is required.**
+
+### The "Resource Group" field.
+
+Having picked a subscription, the resource groups available for that subscription
+will be pre-filled in a drop-down menu. You can either pick one resource group from this menu,
+or click on the "Create new" link right under the drop-down menu to create a new resource group.
+You can read more about Resource groups [here][ResourceGroups].
+
+**This field is required.**
+
+### The "Location" field.
```

Comment:
the location is derived from the resource group in the ARM template, but the portal doesn't seem to understand that: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/keyvault/test-resources.json#L26

Reply:
I'm guessing that's because it's not filled out yet. You could just default it to some name they definitely have to fill in.

So like "fill in a unique name here" or something of that nature.

We can chat about this today if you want. ARM templates are a bit verbose but they have some power behind them.

Reply:
I will make an issue to track this specifically because it needs to be an active effort with Engineering.

Reply:
Issue: https://github.com/Azure/azure-sdk-for-js/issues/7532